internal/exec/subscribe.go

```diff
@@ -154,7 +154,7 @@ func (r *Request) Subscribe(ctx context.Context, s *resolvable.Schema, op *query
 					}()
 
 					if err := subCtx.Err(); err != nil {
-						c <- &Response{Errors: []*errors.QueryError{errors.Errorf("%s", err)}}
+						c <- &Response{Data: nil, Errors: []*errors.QueryError{errors.Errorf("%s", err)}}
```

Comment:
Why do you need to add the `Data: nil`?

Reply:
Hmm looking at that now it does look odd. I think I will need to go back and rethink the stuff in internal/exec/subscribe, now with greater context I am not actually sure how timeouts should really work with subscribe